langlib/langlib-test/src/test/java/org/ballerinalang/langlib/test/LangLibDecimalTest.java

```diff
@@ -216,6 +216,70 @@ public void testRoundAsMethodInvok(BDecimal arg, String expected) {
         };
     }
 
+    @Test(dataProvider = "decimalRoundToFractionDigitsValueProvider")
+    public void testRoundToFraction(BDecimal arg, Integer fractionDigits, String expected) {
+        Object returns = BRunUtil.invoke(compileResult, "testRoundToFractionDigits",
+                new Object[] {arg, fractionDigits});
+        assertEquals(returns, ValueCreator.createDecimalValue(expected));
+    }
+
+    @DataProvider(name = "decimalRoundToFractionDigitsValueProvider")
+    public static Object[][] decimalRoundToFractionDigitsValueProvider() {
+        return new Object[][] {
+                {ValueCreator.createDecimalValue("-504023030303030303030.303567"), 3, "-504023030303030303030.304"},
```

Comment:
I would write these tests in b7a. Creating the values in the Java side ties these tests to jBallerina and even if required we can't directly reuse the tests.

Reply:
We're also relying on the `ValueCreator` more than how a normal user would use it.

Reply:
I will send a separate PR. Not sure if I can meet the deadline with this change.

Reply:
Lets address them in https://github.com/ballerina-platform/ballerina-lang/issues/35756